pages/boards/index.tsx

```diff
@@ -2,24 +2,26 @@ import React from 'react';
 import { useState, useEffect } from 'react';
 import BestPosts from '../../components/BestPosts';
 import AllPosts from '../../components/AllPosts';
-import axios from '../../lib/axios';
 import Image from 'next/image';
-import styles from './index.module.css'
-import {Post} from "@/type/type";
+import styles from './index.module.css';
+import { Post } from '@/type/type';
+import instance from '../../lib/axios';
 
 export default function Home() {
   const [posts, setPosts] = useState<Post[]>([]);
   const [visiblePostsCount, setVisiblePostsCount] = useState(3);
   const [order, setOrder] = useState('recent');
   const [isOpen, setIsOpen] = useState(false);
   const [searchTerm, setSearchTerm] = useState('');
-  const sortedArticles = posts ? [...posts].sort((a, b) => b.likeCount - a.likeCount) : [];
-  const topArticles = sortedArticles.slice(0, visiblePostsCount);
 
+  const sortedArticles = posts
+    ? [...posts].sort((a, b) => b.likeCount - a.likeCount)
+    : [];
```

Comment:
## `posts`의 양이 매우 많아질 수 있다면 메모이제이션을 해볼 수 있어요:
```suggestion
  const sortedArticles = useMemo(() => posts
    ? [...posts].sort((a, b) => b.likeCount - a.likeCount)
    : [], posts)
```

Reply:
### 그리고 서하님께 도움이 될 것 같아서 첨부드려요 !
리액트 메모이제이션에 대한 아티클입니다:
### tl;dr
특히 비용은 useCallback동료 useMemo를 위해 코드를 더 복잡하게 만들고 종속성 배열에서 실수를 할 수 있으며 내장 후크를 호출하고 종속성과 메모된 값을 방지하여 잠재적으로 성능을 저하시킬 수 있다는 것입니다. 가비지 수집으로부터 보호됩니다. 필요한 성능 이점을 얻는 경우 이는 모두 괜찮은 비용이지만 먼저 측정하는 것이 가장 좋습니다.

관련 자료:
- React FAQ: ["렌더링에서 함수를 생성하기 때문에 Hooks가 느려지나요?"](https://reactjs.org/docs/hooks-faq.html#are-hooks-slow-because-of-creating-functions-in-render)
- [Ryan Florence](https://twitter.com/ryanflorence) : [React, 인라인 함수 및 성능](https://reacttraining.com/blog/react-inline-functions-and-performance)

> [원본 보기](https://kentcdodds.com/blog/usememo-and-usecallback)